### ISSUE TITLE
Weird gradle daemon blocking issue

### ISSUE THREAD

**Issue by 8BallBomBom** [safing/portmaster#258]
This is a bit of a weird one to explain but i'll try to cover everything as best as possible, simply put as far as i can tell when running portmaster it seems to be blocking me from starting gradle daemons which are being used to clean or compile android projects.
I can get everything working again easily by stopping the portmaster service or by going into developer mode and switching off process detection.
The way i found this issue was totally by accident and actually caused another issue as well.
I had Android Studio running in the background trying to clean + compile and for some reason due to this blocking issue loads of java processes were created but also left to hang in the background.
The result of that was all cpu cores and memory on the system being consumed almost entirely.
After getting things under control i then found that portmaster freaked which resulted in all of my cpu being consumed again.
It would seem portmaster doesn't like it if you have tons of processes created and then hung in the background in quick succession.

Anywho here is a stack trace from trying to run a gradle clean when portmaster is running.
```
./gradlew clean --stacktrace
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/6.5/userguide/gradle_daemon.html.

FAILURE: Build failed with an exception.

* What went wrong:
The first result from the daemon was empty. Most likely the process died immediately after connection.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.launcher.daemon.client.DaemonInitialConnectException: The first result from the daemon was empty. Most likely the process died immediately after connection.
        at org.gradle.launcher.daemon.client.DaemonClient.executeBuild(DaemonClient.java:177)
        at org.gradle.launcher.daemon.client.SingleUseDaemonClient.execute(SingleUseDaemonClient.java:55)
        at org.gradle.launcher.daemon.client.SingleUseDaemonClient.execute(SingleUseDaemonClient.java:38)
        at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:56)
        at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:212)
        at org.gradle.launcher.cli.DefaultCommandLineActionFactory$ParseAndBuildAction.execute(DefaultCommandLineActionFactory.java:294)
        at org.gradle.launcher.cli.DefaultCommandLineActionFactory$ParseAndBuildAction.execute(DefaultCommandLineActionFactory.java:266)
        at org.gradle.launcher.cli.DebugLoggerWarningAction.execute(DebugLoggerWarningAction.java:82)
        at org.gradle.launcher.cli.DebugLoggerWarningAction.execute(DebugLoggerWarningAction.java:30)
        at org.gradle.launcher.cli.WelcomeMessageAction.execute(WelcomeMessageAction.java:92)
        at org.gradle.launcher.cli.WelcomeMessageAction.execute(WelcomeMessageAction.java:38)
        at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:44)
        at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:26)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:41)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:26)
        at org.gradle.launcher.cli.DefaultCommandLineActionFactory$WithLogging.execute(DefaultCommandLineActionFactory.java:259)
        at org.gradle.launcher.Main.doAction(Main.java:35)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:50)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:60)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:37)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:31)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:35)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:108)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:63)


* Get more help at https://help.gradle.org
```
After looking through some logs, when the gradle daemon starts i can see an obvious difference.
With portmaster disabled it creates services on 127.0.0.1 but with portmaster running it instead creates services on 0.0.0.0 which seems very odd, but definitely can see straight after that point everything fails.
I can try to provide some actual proper logs if needed, just trying to go through some of them currently.
Here is also some info from the portmaster debug log if you need it.
```
Portmaster
version 0.6.5

commit tags/v0.6.5-0-g6c0df1155e5bc19e8b749875ef984c34eeb8b315
built with go1.15.8 (gc) linux/amd64
  using options main.go
  by user@docker
  on 11.02.2021

Platform: ubuntu 20.10
System: ubuntu linux (debian) 20.10
Kernel: 5.10.16-xanmod1 x86_64
```

**Comment by davegson** [comment]
Hey there, yesterday and today we were heavily focused on other things, so sadly a proper inspection and response to the issue will still have to wait.

Nevertheless, thanks a lot for the report - we appreciate it! Expect a proper response next week. Cheers!

**Comment by 8BallBomBom** [comment]
No worries.
I'll try and add more debug logs soon.

**Comment by dhaavi** [comment]
Hey @8BallBomBom, thanks for reporting this.

This seems rather weird that a program would change how it binds a listener. I may be doing a network check first to find out what it should do. That could cause the problem.

In order to take a closer look at this, please share the trace logs from the Portmaster from when this is happening:
```
sudo systemctl stop portmaster # stop current instance
sudo $(/var/lib/portmaster/portmaster-start show core) --log trace # start the current portmaster version with trace logging
# then copy the part where Android Studio is having an issue
sudo systemctl start portmaster # resume normal background operation when done
```

**Comment by 8BallBomBom** [comment]
Hope this is good enough, some ip addresses in there but no biggie.
```
210222 21:18:40.818 connection:564 ▶ INFO 263 filter: connection Unknown::-1 <- 136.57.185.238 dropped: incoming connection blocked by default Σ=164.353839ms
             5.45µs terception:101 ▶ TRAC     filter: handling packet: IN TCP 192.168.0.10:6881 <-> 136.57.185.238:33216
       164.216969ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
            16.65µs ocess/find:026 ▶ DEBU     process: failed to find PID of connection: could not find connection in system state tables
            10.95µs connection:272 ▶ DEBU     network: failed to find process of packet IN TCP 192.168.0.10:6881 <-> 136.57.185.238:33216: could not find connection in system state tables
            40.18µs terception:111 ▶ TRAC     filter: created new connection 6-192.168.0.10-6881-136.57.185.238-33216
             3.43µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
            60.21µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:40.818 on/nfq/nfq:194 ▶ TRAC 264 nfqueue: queued packet pkt:54 qid:17140 (92.29.216.126 -> 192.168.0.10) after 15.88µs
210222 21:18:40.818 nfq/packet:116 ▶ TRAC 265 nfqueue: marking packet pkt:54 qid:17140 (92.29.216.126 -> 192.168.0.10) on queue 17140 with DropAlways after 85.26µs
210222 21:18:40.912 on/nfq/nfq:194 ▶ TRAC 266 nfqueue: queued packet pkt:3 qid:17160 (2001:470:e0cf:a357:a00:27ff:fed3:7266 -> 2001:4454:3ba:1a00:87e:77e1:2880:749b) after 21.71µs
210222 21:18:40.987 on/nfq/nfq:194 ▶ TRAC 267 nfqueue: queued packet pkt:55 qid:17140 (72.68.207.129 -> 192.168.0.10) after 20.41µs
210222 21:18:40.987 nfq/packet:116 ▶ TRAC 268 nfqueue: marking packet pkt:55 qid:17140 (72.68.207.129 -> 192.168.0.10) on queue 17140 with DropAlways after 119.629µs
210222 21:18:41.005 on/nfq/nfq:194 ▶ TRAC 269 nfqueue: queued packet pkt:1 qid:17060 (::1 -> ::1) after 21.59µs
210222 21:18:41.055 on/nfq/nfq:194 ▶ TRAC 270 nfqueue: queued packet pkt:11 qid:17040 (127.0.0.1 -> 127.0.0.1) after 18.97µs
210222 21:18:41.084 nfq/packet:116 ▶ TRAC 271 nfqueue: marking packet pkt:2 qid:17160 (2001:470:e0cf:a357:a00:27ff:fed3:7266 -> 2001:4454:3ba:1a00:87e:77e1:2880:749b) on queue 17160 with AcceptAlways after 1.193280222s
210222 21:18:41.084 connection:564 ▶ INFO 272 filter: connection Unknown::-1 -> 2001:4454:3ba:1a00:87e:77e1:2880:749b accepted: default permit Σ=1.193279093s
             3.41µs terception:101 ▶ TRAC     filter: handling packet: IN TCP 2001:4454:3ba:1a00:87e:77e1:2880:749b:22556 <-> 2001:470:e0cf:a357:a00:27ff:fed3:7266:59538
       1.193098583s ocess/find:022 ▶ TRAC     process: getting pid from system network state
             7.56µs ss/process:095 ▶ TRAC     process: getting process for PID -1
             30.1µs ss/profile:026 ▶ TRAC     process: profile already loaded
            29.34µs terception:111 ▶ TRAC     filter: created new connection 6-2001:4454:3ba:1a00:87e:77e1:2880:749b-22556-2001:470:e0cf:a357:a00:27ff:fed3:7266-59538
             3.71µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
           106.39µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:41.128 on/nfq/nfq:194 ▶ TRAC 273 nfqueue: queued packet pkt:2 qid:17060 (::1 -> ::1) after 25.39µs
210222 21:18:41.164 nfq/packet:116 ▶ TRAC 274 nfqueue: marking packet pkt:1 qid:17060 (::1 -> ::1) on queue 17060 with AcceptAlways after 159.413594ms
210222 21:18:41.164 connection:564 ▶ INFO 275 filter: connection johnathon:/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33466 -> ::1 accepted: permitted by rule: scope matches Localhost Σ=159.339304ms
             7.54µs terception:101 ▶ TRAC     filter: handling packet: OUT TCP ::1:55424 <-> ::1:7
         4.958116ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
           252.62µs ss/process:095 ▶ TRAC     process: getting process for PID 33466
           108.02µs ss/profile:031 ▶ TRAC     process: loading profile
            20.05µs terception:111 ▶ TRAC     filter: created new connection 6-::1-55424-::1-7
             3.15µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
       153.886918ms terception:280 ▶ TRAC     filter: starting decision process
           102.89µs all/master:150 ▶ WARN     filter: failed to find local peer process PID: could not find connection in system state tables
210222 21:18:41.178 on/nfq/nfq:194 ▶ TRAC 277 nfqueue: queued packet pkt:12 qid:17040 (127.0.0.1 -> 127.0.0.1) after 22.07µs
210222 21:18:41.239 on/nfq/nfq:194 ▶ TRAC 278 nfqueue: queued packet pkt:13 qid:17040 (127.0.0.1 -> 127.0.1.1) after 77.67µs
210222 21:18:41.325 nfq/packet:116 ▶ TRAC 279 nfqueue: marking packet pkt:11 qid:17040 (127.0.0.1 -> 127.0.0.1) on queue 17040 with AcceptAlways after 269.747882ms
210222 21:18:41.325 connection:564 ▶ INFO 280 filter: connection johnathon:/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33466 -> 127.0.0.1 accepted: permitted by rule: scope matches Localhost Σ=269.699172ms
             6.02µs terception:101 ▶ TRAC     filter: handling packet: OUT TCP 127.0.0.1:60536 <-> 127.0.0.1:7
         5.290085ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             5.22µs ss/process:095 ▶ TRAC     process: getting process for PID 33466
       103.541385ms ss/profile:026 ▶ TRAC     process: profile already loaded
             18.2µs terception:111 ▶ TRAC     filter: created new connection 6-127.0.0.1-60536-127.0.0.1-7
             3.28µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
       160.735452ms terception:280 ▶ TRAC     filter: starting decision process
            99.53µs all/master:150 ▶ WARN     filter: failed to find local peer process PID: could not find connection in system state tables
210222 21:18:41.478 nfq/packet:116 ▶ TRAC 282 nfqueue: marking packet pkt:2 qid:17060 (::1 -> ::1) on queue 17060 with AcceptAlways after 349.865667ms
210222 21:18:41.478 connection:564 ▶ INFO 283 filter: connection johnathon:/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33432 -> ::1 accepted: permitted by rule: scope matches Localhost Σ=349.784877ms
              9.5µs terception:101 ▶ TRAC     filter: handling packet: OUT TCP ::1:55428 <-> ::1:7
         5.046214ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
           216.95µs ss/process:095 ▶ TRAC     process: getting process for PID 33432
       191.621254ms ss/profile:031 ▶ TRAC     process: loading profile
            33.96µs terception:111 ▶ TRAC     filter: created new connection 6-::1-55428-::1-7
             5.98µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
       152.749269ms terception:280 ▶ TRAC     filter: starting decision process
           101.75µs all/master:150 ▶ WARN     filter: failed to find local peer process PID: could not find connection in system state tables
210222 21:18:41.608 on/nfq/nfq:194 ▶ TRAC 285 nfqueue: queued packet pkt:56 qid:17140 (109.41.65.32 -> 192.168.0.10) after 23.73µs
210222 21:18:41.640 nfq/packet:116 ▶ TRAC 286 nfqueue: marking packet pkt:12 qid:17040 (127.0.0.1 -> 127.0.0.1) on queue 17040 with AcceptAlways after 461.276856ms
210222 21:18:41.640 connection:564 ▶ INFO 287 filter: connection johnathon:/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33432 -> 127.0.0.1 accepted: permitted by rule: scope matches Localhost Σ=461.223016ms
             6.76µs terception:101 ▶ TRAC     filter: handling packet: OUT TCP 127.0.0.1:60540 <-> 127.0.0.1:7
         2.015078ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
       144.768407ms ss/process:095 ▶ TRAC     process: getting process for PID 33432
            34.86µs ss/profile:026 ▶ TRAC     process: profile already loaded
            22.56µs terception:111 ▶ TRAC     filter: created new connection 6-127.0.0.1-60540-127.0.0.1-7
             3.81µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
       314.249881ms terception:280 ▶ TRAC     filter: starting decision process
           121.66µs all/master:150 ▶ WARN     filter: failed to find local peer process PID: could not find connection in system state tables
210222 21:18:41.640 nfq/packet:116 ▶ TRAC 289 nfqueue: marking packet pkt:13 qid:17040 (127.0.0.1 -> 127.0.1.1) on queue 17040 with AcceptAlways after 400.531182ms
210222 21:18:41.640 connection:564 ▶ INFO 290 filter: connection johnathon:/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33432 -> 127.0.1.1 accepted: permitted by rule: scope matches Localhost Σ=400.409352ms
             6.57µs terception:101 ▶ TRAC     filter: handling packet: OUT TCP 127.0.0.1:53764 <-> 127.0.1.1:40531
         6.029974ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
        79.935637ms ss/process:095 ▶ TRAC     process: getting process for PID 33432
       314.330521ms ss/profile:026 ▶ TRAC     process: profile already loaded
            18.73µs terception:111 ▶ TRAC     filter: created new connection 6-127.0.0.1-53764-127.0.1.1-40531
             3.96µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
            83.96µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:41.658 on/nfq/nfq:194 ▶ TRAC 291 nfqueue: queued packet pkt:57 qid:17140 (136.57.185.238 -> 192.168.0.10) after 17.51µs
210222 21:18:41.658 nfq/packet:116 ▶ TRAC 292 nfqueue: marking packet pkt:57 qid:17140 (136.57.185.238 -> 192.168.0.10) on queue 17140 with DropAlways after 91.05µs
210222 21:18:41.765 on/nfq/nfq:194 ▶ TRAC 293 nfqueue: queued packet pkt:58 qid:17140 (112.204.162.179 -> 192.168.0.10) after 22.41µs
210222 21:18:41.765 nfq/packet:116 ▶ TRAC 294 nfqueue: marking packet pkt:58 qid:17140 (112.204.162.179 -> 192.168.0.10) on queue 17140 with DropAlways after 98.31µs
210222 21:18:41.776 on/nfq/nfq:194 ▶ TRAC 295 nfqueue: queued packet pkt:59 qid:17140 (188.166.62.180 -> 192.168.0.10) after 14.82µs
210222 21:18:41.870 nfq/packet:116 ▶ TRAC 296 nfqueue: marking packet pkt:49 qid:17140 (109.41.65.32 -> 192.168.0.10) on queue 17140 with AcceptAlways after 1.263389727s
210222 21:18:41.870 connection:564 ▶ INFO 297 filter: connection Unknown::-1 -> 192.168.0.10 accepted: default permit Σ=1.263377907s
             4.57µs terception:101 ▶ TRAC     filter: handling packet: IN TCP 192.168.0.10:22556 <-> 109.41.65.32:62018
       1.263173898s ocess/find:022 ▶ TRAC     process: getting pid from system network state
            16.26µs ss/process:095 ▶ TRAC     process: getting process for PID -1
            29.92µs ss/profile:026 ▶ TRAC     process: profile already loaded
            35.42µs terception:111 ▶ TRAC     filter: created new connection 6-192.168.0.10-22556-109.41.65.32-62018
             3.91µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
          113.929µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:41.906 on/nfq/nfq:194 ▶ TRAC 298 nfqueue: queued packet pkt:60 qid:17140 (94.66.225.195 -> 192.168.0.10) after 27.13µs
210222 21:18:41.993 nfq/packet:116 ▶ TRAC 299 nfqueue: marking packet pkt:53 qid:17140 (188.166.62.180 -> 192.168.0.10) on queue 17140 with AcceptAlways after 1.238363299s
210222 21:18:41.993 connection:564 ▶ INFO 300 filter: connection Unknown::-1 -> 192.168.0.10 accepted: default permit Σ=1.238320599s
             7.29µs terception:101 ▶ TRAC     filter: handling packet: IN TCP 192.168.0.10:22556 <-> 188.166.62.180:7095
        1.23801365s ocess/find:022 ▶ TRAC     process: getting pid from system network state
             6.52µs ss/process:095 ▶ TRAC     process: getting process for PID -1
            26.36µs ss/profile:026 ▶ TRAC     process: profile already loaded
            19.36µs terception:111 ▶ TRAC     filter: created new connection 6-192.168.0.10-22556-188.166.62.180-7095
             2.76µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
          244.659µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:42.026 on/nfq/nfq:194 ▶ TRAC 301 nfqueue: queued packet pkt:14 qid:17040 (127.0.0.1 -> 127.0.0.53) after 23.75µs
210222 21:18:42.026 on/nfq/nfq:194 ▶ TRAC 302 nfqueue: queued packet pkt:15 qid:17040 (127.0.0.1 -> 127.0.0.53) after 2.99µs
210222 21:18:42.029 nfq/packet:116 ▶ TRAC 303 nfqueue: marking packet pkt:15 qid:17040 (127.0.0.1 -> 127.0.0.53) on queue 17040 with AcceptAlways after 3.519136ms
210222 21:18:42.029 connection:564 ▶ INFO 304 filter: connection root:/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-6-5:33337 -> 127.0.0.53 accepted: connection by Portmaster Σ=3.492096ms
             5.55µs terception:101 ▶ TRAC     filter: handling packet: OUT UDP 127.0.0.1:60812 <-> 127.0.0.53:53
         3.368777ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             3.73µs ss/process:095 ▶ TRAC     process: getting process for PID 33337
            19.81µs ss/profile:026 ▶ TRAC     process: profile already loaded
             14.3µs terception:111 ▶ TRAC     filter: created new connection 17-127.0.0.1-60812-127.0.0.53-53
             3.35µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
            14.06µs terception:280 ▶ TRAC     filter: starting decision process
           62.519µs all/master:123 ▶ INFO     filter: granting own connection root:/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-6-5:33337 -> 127.0.0.53
210222 21:18:42.029 nfq/packet:116 ▶ TRAC 305 nfqueue: marking packet pkt:14 qid:17040 (127.0.0.1 -> 127.0.0.53) on queue 17040 with AcceptAlways after 3.587826ms
210222 21:18:42.029 connection:564 ▶ INFO 306 filter: connection root:/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-6-5:33337 -> 127.0.0.53 accepted: connection by Portmaster Σ=3.474466ms
             6.34µs terception:101 ▶ TRAC     filter: handling packet: OUT UDP 127.0.0.1:60812 <-> 127.0.0.53:53
         3.342327ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             2.24µs ss/process:095 ▶ TRAC     process: getting process for PID 33337
            12.27µs ss/profile:026 ▶ TRAC     process: profile already loaded
            13.53µs terception:111 ▶ TRAC     filter: created new connection 17-127.0.0.1-60812-127.0.0.53-53
             3.15µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
            14.41µs terception:280 ▶ TRAC     filter: starting decision process
           80.199µs all/master:123 ▶ INFO     filter: granting own connection root:/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-6-5:33337 -> 127.0.0.53
210222 21:18:42.029 on/nfq/nfq:194 ▶ TRAC 307 nfqueue: queued packet pkt:16 qid:17040 (192.168.0.10 -> 1.1.1.1) after 8.81µs
210222 21:18:42.029 on/nfq/nfq:194 ▶ TRAC 308 nfqueue: queued packet pkt:17 qid:17040 (192.168.0.10 -> 1.1.1.1) after 2.62µs
210222 21:18:42.033 nfq/packet:116 ▶ TRAC 309 nfqueue: marking packet pkt:16 qid:17040 (192.168.0.10 -> 1.1.1.1) on queue 17040 with RerouteNS after 3.553767ms
210222 21:18:42.033 connection:564 ▶ INFO 310 filter: connection systemd-resolve:/usr/lib/systemd/systemd-resolved:9734 to one.one.one.one. (1.1.1.1) rerouted to nameserver: redirecting rogue dns query Σ=3.542877ms
              3.1µs terception:101 ▶ TRAC     filter: handling packet: OUT UDP 192.168.0.10:34228 <-> 1.1.1.1:53
         3.432347ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             3.95µs ss/process:095 ▶ TRAC     process: getting process for PID 9734
            33.18µs ss/profile:026 ▶ TRAC     process: profile already loaded
            12.46µs terception:111 ▶ TRAC     filter: created new connection 17-192.168.0.10-34228-1.1.1.1-53
            57.84µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
210222 21:18:42.033 nfq/packet:116 ▶ TRAC 311 nfqueue: marking packet pkt:17 qid:17040 (192.168.0.10 -> 1.1.1.1) on queue 17040 with RerouteNS after 3.575317ms
210222 21:18:42.033 connection:564 ▶ INFO 312 filter: connection systemd-resolve:/usr/lib/systemd/systemd-resolved:9734 to one.one.one.one. (1.1.1.1) rerouted to nameserver: redirecting rogue dns query Σ=3.576367ms
             2.09µs terception:101 ▶ TRAC     filter: handling packet: OUT UDP 192.168.0.10:37110 <-> 1.1.1.1:53
         3.452877ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             3.69µs ss/process:095 ▶ TRAC     process: getting process for PID 9734
            34.54µs ss/profile:026 ▶ TRAC     process: profile already loaded
            21.35µs terception:111 ▶ TRAC     filter: created new connection 17-192.168.0.10-37110-1.1.1.1-53
            61.82µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
210222 21:18:42.033 nameserver:239 ▶ INFO 313 nameserver: returning accepted response (NOERROR) for updates.safing.io.AAAA to systemd-resolve:/usr/lib/systemd/systemd-resolved:9734 Σ=156.02µs
            12.76µs nameserver:064 ▶ TRAC     nameserver: handling new request for updates.safing.io.AAAA from 192.168.0.10:34228
              2.1µs ocess/find:022 ▶ TRAC     process: getting pid from system network state
             2.63µs ss/process:095 ▶ TRAC     process: getting process for PID 9734
            29.87µs ss/profile:026 ▶ TRAC     process: profile already loaded
             4.56µs all/master:355 ▶ TRAC     filter: LMS score of eTLD+1 safing.io is 100.00
            92.53µs er/resolve:127 ▶ TRAC     resolver: resolving updates.safing.io.AAAA
             3.27µs er/resolve:220 ▶ TRAC     resolver: cache for updates.safing.io.AAAA will expire in 7s, refreshing async now
              8.3µs nameserver:205 ▶ TRAC     nameserver: deciding on resolved dns
210222 21:18:42.033 nameserver:239 ▶ INFO 314 nameserver: returning accepted response (NOERROR) for updates.safing.io.A to systemd-resolve:/usr/lib/systemd/systemd-resolved:9734 Σ=132.39µs
             8.12µs nameserver:064 ▶ TRAC     nameserver: handling new request for updates.safing.io.A from 192.168.0.10:37110
             1.68µs ocess/find:022 ▶ TRAC     process: getting pid from system network state
             2.34µs ss/process:095 ▶ TRAC     process: getting process for PID 9734
             7.74µs ss/profile:026 ▶ TRAC     process: profile already loaded
             2.81µs all/master:355 ▶ TRAC     filter: LMS score of eTLD+1 safing.io is 100.00
            83.54µs er/resolve:127 ▶ TRAC     resolver: resolving updates.safing.io.A
              4.4µs er/resolve:220 ▶ TRAC     resolver: cache for updates.safing.io.A will expire in 7s, refreshing async now
            21.76µs nameserver:205 ▶ TRAC     nameserver: deciding on resolved dns
210222 21:18:42.033 on/nfq/nfq:194 ▶ TRAC 315 nfqueue: queued packet pkt:18 qid:17040 (192.168.0.10 -> 1.1.1.2) after 15.39µs
210222 21:18:42.033 on/nfq/nfq:194 ▶ TRAC 316 nfqueue: queued packet pkt:19 qid:17040 (192.168.0.10 -> 1.1.1.1) after 7.15µs
210222 21:18:42.037 nfq/packet:116 ▶ TRAC 317 nfqueue: marking packet pkt:19 qid:17040 (192.168.0.10 -> 1.1.1.1) on queue 17040 with RerouteNS after 3.880086ms
210222 21:18:42.037 connection:564 ▶ INFO 318 filter: connection systemd-resolve:/usr/lib/systemd/systemd-resolved:9734 to one.one.one.one. (1.1.1.1) rerouted to nameserver: redirecting rogue dns query Σ=3.860356ms
             4.39µs terception:101 ▶ TRAC     filter: handling packet: OUT UDP 192.168.0.10:54124 <-> 1.1.1.1:53
         3.728677ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             4.43µs ss/process:095 ▶ TRAC     process: getting process for PID 9734
            31.08µs ss/profile:026 ▶ TRAC     process: profile already loaded
           16.939µs terception:111 ▶ TRAC     filter: created new connection 17-192.168.0.10-54124-1.1.1.1-53
            74.84µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
210222 21:18:42.038 nfq/packet:116 ▶ TRAC 319 nfqueue: marking packet pkt:18 qid:17040 (192.168.0.10 -> 1.1.1.2) on queue 17040 with AcceptAlways after 5.111275ms
210222 21:18:42.038 connection:564 ▶ INFO 320 filter: connection root:/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-6-5:33337 -> 1.1.1.2 accepted: connection by Portmaster Σ=5.041815ms
             2.19µs terception:101 ▶ TRAC     filter: handling packet: OUT TCP 192.168.0.10:56983 <-> 1.1.1.2:853
         4.968665ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
             2.35µs ss/process:095 ▶ TRAC     process: getting process for PID 33337
             9.86µs ss/profile:026 ▶ TRAC     process: profile already loaded
             13.1µs terception:111 ▶ TRAC     filter: created new connection 6-192.168.0.10-56983-1.1.1.2-853
            45.65µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
210222 21:18:42.052 terception:416 ▶ TRAC 321 filter: packets accepted 31, blocked 0, dropped 43, failed 0
210222 21:18:42.066 nfq/packet:116 ▶ TRAC 322 nfqueue: marking packet pkt:60 qid:17140 (94.66.225.195 -> 192.168.0.10) on queue 17140 with DropAlways after 159.825323ms
210222 21:18:42.066 connection:564 ▶ INFO 323 filter: connection Unknown::-1 <- 94.66.225.195 dropped: incoming connection blocked by default Σ=159.475653ms
              8.8µs terception:101 ▶ TRAC     filter: handling packet: IN TCP 192.168.0.10:6881 <-> 94.66.225.195:47578
       159.354083ms ocess/find:022 ▶ TRAC     process: getting pid from system network state
            21.82µs ocess/find:026 ▶ DEBU     process: failed to find PID of connection: could not find connection in system state tables
             8.41µs connection:272 ▶ DEBU     network: failed to find process of packet IN TCP 192.168.0.10:6881 <-> 94.66.225.195:47578: could not find connection in system state tables
            22.98µs terception:111 ▶ TRAC     filter: created new connection 6-192.168.0.10-6881-94.66.225.195-47578
             3.03µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
            56.53µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:42.135 nfq/packet:116 ▶ TRAC 324 nfqueue: marking packet pkt:3 qid:17160 (2001:470:e0cf:a357:a00:27ff:fed3:7266 -> 2001:4454:3ba:1a00:87e:77e1:2880:749b) on queue 17160 with AcceptAlways after 1.222535695s
210222 21:18:42.135 connection:564 ▶ INFO 325 filter: connection Unknown::-1 -> 2001:4454:3ba:1a00:87e:77e1:2880:749b accepted: default permit Σ=1.222480784s
            8.019µs terception:101 ▶ TRAC     filter: handling packet: IN TCP 2001:4454:3ba:1a00:87e:77e1:2880:749b:22556 <-> 2001:470:e0cf:a357:a00:27ff:fed3:7266:59538
       1.222321896s ocess/find:022 ▶ TRAC     process: getting pid from system network state
             6.22µs ss/process:095 ▶ TRAC     process: getting process for PID -1
            25.53µs ss/profile:026 ▶ TRAC     process: profile already loaded
            18.75µs terception:111 ▶ TRAC     filter: created new connection 6-2001:4454:3ba:1a00:87e:77e1:2880:749b-22556-2001:470:e0cf:a357:a00:27ff:fed3:7266-59538
             2.81µs terception:243 ▶ TRAC     filter: handing over to connection-based handler
           97.559µs terception:280 ▶ TRAC     filter: starting decision process
210222 21:18:42.432 on/nfq/nfq:194 ▶ TRAC 326 nfqueue: queued packet pkt:61 qid:17140 (99.235.83.75 -> 192.168.0.10) after 16.63µs
210222 21:18:42.432 nfq/packet:116 ▶ TRAC 327 nfqueue: marking packet pkt:61 qid:17140 (99.235.83.75 -> 192.168.0.10) on queue 17140 with DropAlways after 90.57µs
^C <INTERRUPT>
210222 21:18:42.624 4/run/main:075 ▶ WARN 328 main: program was interrupted, shutting down.
210222 21:18:42.624 dules/stop:043 ▶ WARN 330 modules: starting shutdown...
```

**Comment by 8BallBomBom** [comment]
Failed android cleaning build process when running portmaster, initially gives this.
```
2021-02-22T21:24:03.789+0800 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: DETACHED
2021-02-22T21:24:03.789+0800 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Process 'Gradle build daemon' finished with exit value 0 (state: DETACHED)
2021-02-22T21:24:03.789+0800 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] Gradle daemon process is now detached.
2021-02-22T21:24:03.791+0800 [INFO] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] An attempt to start the daemon took 0.602 secs.
2021-02-22T21:24:03.792+0800 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonConnector] Started Gradle daemon DaemonStartupInfo{pid=33904, uid=f8f7793c-aa9b-4cff-86ea-fc9a9b950289, address=[879d412e-e264-4793-b88a-f27fa2f2f212 port:44267, addresses:[/0.0.0.0]]
```
But without portmaster it shows this and happily cleans then compiles.
```
2021-02-22T21:27:38.564+0800 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: DETACHED
2021-02-22T21:27:38.564+0800 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Process 'Gradle build daemon' finished with exit value 0 (state: DETACHED)
2021-02-22T21:27:38.564+0800 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] Gradle daemon process is now detached.
2021-02-22T21:27:38.567+0800 [INFO] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] An attempt to start the daemon took 0.503 secs.
2021-02-22T21:27:38.567+0800 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonConnector] Started Gradle daemon DaemonStartupInfo{pid=34118, uid=9e56bffc-9c15-4974-bc52-06432722d5e3, address=[964e9db8-283d-4a74-b026-fadd191c0a53 port:45205, addresses:[/127.0.0.1]]
```

**Comment by dhaavi** [comment]
Thanks for the logs!

It is doing pretty much what I expected. It's checking network capabilities, trying to connect to the local port 7:
```
/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33466 -> ::1 accepted: permitted by rule: scope matches Localhost Σ=159.339304ms
handling packet: OUT TCP ::1:55424 <-> ::1:7

/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33466 -> 127.0.0.1 accepted: permitted by rule: scope matches Localhost Σ=269.699172ms
handling packet: OUT TCP 127.0.0.1:60536 <-> 127.0.0.1:7

/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33432 -> ::1 accepted: permitted by rule: scope matches Localhost Σ=349.784877ms
handling packet: OUT TCP ::1:55428 <-> ::1:7

/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33432 -> 127.0.0.1 accepted: permitted by rule: scope matches Localhost Σ=461.223016ms
handling packet: OUT TCP 127.0.0.1:60540 <-> 127.0.0.1:7
```

Though I am not sure what the fifth connection is doing. Maybe there is something running on `127.0.1.1:40531`?
```
/usr/lib/jvm/java-14-openjdk-amd64/bin/java:33432 -> 127.0.1.1 accepted: permitted by rule: scope matches Localhost Σ=400.409352ms
handling packet: OUT TCP 127.0.0.1:53764 <-> 127.0.1.1:40531
```

It could be that the Portmaster is too slow and that the program thinks that the connection is failing. The majority of the handling time spent - in this case - is to check whether there actually is something on the other end of that connection. Maybe we should do this a lot less agressive.
```
filter: failed to find local peer process PID: could not find connection in system state tables
```

The fact that switching off process detection also solves this issue somewhat proves that speed is the problem here. I will take measures to improve that!

**Comment by dhaavi** [comment]
This was closed in error, sorry. We normally close issue only after confirmation that the fix works.

Anyway, this might now be fixed with [v1.0.9 of the Portmaster Kernel Extension](https://github.com/safing/portmaster-windows-kext/releases/tag/v1.0.9). If you are not reinstalling, please make sure you have v1.0.9 of the Kext by looking it up in the version details on the settings page.

**Comment by 8BallBomBom** [comment]
Sadly that didn't make any difference, running an updated unstable release with updated Kext but sadly the same issue occurs.

**Comment by dhaavi** [comment]
With [v0.6.7](https://github.com/safing/portmaster/releases/tag/v0.6.7), we were able to work around the requirement that the Portmaster needs the Windows DNS-Client service stopped. The Portmaster now re-enables the DNS-Client and the new installer does not even disable it anymore now.

This might also have an effect on this issue. If you find the time to test this, please report back.

**Comment by 8BallBomBom** [comment]
I'm not running windows, not sure how that applies as i'm running ubuntu.

**Comment by dhaavi** [comment]
Ah, yes. :facepalm: 

You are right, in this case this should not have any impact. I somehow thought that this was also waiting for this issue to be resolved - sorry for letting this issue hang, then.

Ok. We have a new button to collect some information, could you press that when you run into the issue again and post the data here? It's in the monitor view of an application in the dots-menu on the far right of the app information section header and is called "Copy Debug Information". This will provide us with a couple more details to look into this issue.

**Comment by 8BallBomBom** [comment]
No worries i'll get back to you soon with some more debug information 👍🏻

**Comment by 8BallBomBom** [comment]
I'll have to attempt this a little later, the connection from the Philippines to Europe isn't exactly working decent currently.
Max i'm getting downloading the client is around 12kb/s and after that took a while to download the actual installer takes a long while to even get started.

**Comment by dhaavi** [comment]
[Slightly off topic]

> the connection from the Philippines to Europe isn't exactly working decent currently

That is really interesting. I wonder if we'd be able to find a faster route using the SPN. This is something I'd like to look into in a couple months time when you're interested too.

**Comment by 8BallBomBom** [comment]
I'd be interested in testing if you'd need that of-course 👍🏻 
Now that things have started to clean up i'll get a debug log posted here shortly.

**Comment by 8BallBomBom** [comment]
It would seem we have a bit of a rogue situation going on.
After a total purge and reinstall of the latest stable version, everything seems to be working as expected.
Possibly the updated v1.0.9 of the Portmaster Kernel Extension might have fixed it?
But previously i forced unstable updates and confirmed that i was running that version.
Along with a few purges and re-installs without any luck and the same issue occurring.

So now all i can figure is something included in a recent update in the stable build has resolved the issue.

If you'd still like me to grab debug logs from any of the processes that were having issues by all means give me a nudge 😃

**Comment by dhaavi** [comment]
> After a total purge and reinstall of the latest stable version, everything seems to be working as expected.

Great!

> Possibly the updated v1.0.9 of the Portmaster Kernel Extension might have fixed it?

The Kext is Windows only, so that should not have affected this. There might have been a problem with the iptables integration, but there was no change there.

> But previously i forced unstable updates and confirmed that i was running that version.

What kind of unstable updates are you referring to?

> If you'd still like me to grab debug logs from any of the processes that were having issues by all means give me a nudge

Only if the problem appears again - but thank you for the offer!

**Comment by 8BallBomBom** [comment]
When i said unstable updates i was referring to running unstable versions to gain access to the Kext update.
Thanks for the help, if i have any issues i'll report them immediately 👍🏻
